packages/firestore/src/core/bundle.ts

```diff
@@ -244,10 +244,12 @@ export class BundleLoader {
         !!this.documents[this.documents.length - 1].document,
       'Bundled documents ends with a document metadata and missing document.'
     );
+    debugAssert(!!this.metadata.id, 'Bundle ID must be set.');
 
     const changedDocuments = await applyBundleDocuments(
       this.localStore,
-      this.documents
+      this.documents,
+      this.metadata.id!
```

Comment:
You mean renaming the proto?

Reply:
Hm, then maybe not. Should we settle on "bundleId" then? Feel free to push back or address later.

Reply:
I think `id` makes sense when viewed as a field in `BundleMetadata`. Here it is a little vague though, maybe it will be slightly better to rename it to `bundleMetadata`?

Reply:
You can leave as is. I think `metadata` or `bundleMetadata` doesn't make a huge difference since this module is all about bundles.